### PR TITLE
Only resolve aliases when emitting typedefs.

### PR DESCRIPTION
**Pull request by mprobst**
This is based on PR #654, please only review the last commit.

### CODE REVIEW

File: src/type-translator.ts [hunk 1]
```diff
@@ -125,10 +125,10 @@ export function symbolToDebugString(sym: ts.Symbol): string {
 /** TypeTranslator translates TypeScript types to Closure types. */
 export class TypeTranslator {
   /**
-   * A list of types we've encountered while emitting; used to avoid getting stuck in recursive
-   * types.
+   * A list of type literals we've encountered while emitting; used to avoid getting stuck in
+   * recursive types.
    */
-  private readonly seenTypes: ts.Type[] = [];
+  private readonly seenTypeLiterals: ts.Type[] = [];
```
Comment (evmar) [inline src/type-translator.ts]: You can use a Set<> here.

File: test_files/type_alias_imported/type_alias_exporter.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export class X { private x: string; }
+export class Y { private x: string; }
+export type XY = X|Y;
```
Comment (evmar) [inline test_files/type_alias_imported/type_alias_exporter.ts]: Docs in here about what this is looking for are again helpful.

File: src/type-translator.ts (hunk 1 above)
Comment (mprobst) [reply src/type-translator.ts]: I stumbled over that, and was wondering if you used an array on purpose given these are very likely never more than one or two types.

File: test_files/type_alias_imported/type_alias_exporter.ts (hunk 2 above)
Comment (mprobst) [reply test_files/type_alias_imported/type_alias_exporter.ts]: Done.

File: src/tsickle.ts [hunk 3]
```diff
@@ -1128,6 +1128,18 @@ class Annotator extends ClosureRewriter {
...
+      // module because it's only used in type positions, the JSDoc comments still reference a valid
+      // Closure level symbol.
+      const sym = this.typeChecker.getSymbolAtLocation(decl.moduleSpecifier);
+      if (!sym) {
+        return true;  // modules might not have a symbol if they are unused.
+      }
+      // forwardDeclare all symbols that can be imported from the module.
+      const namedSyms =
+          this.typeChecker.getExportsOfModule(sym).map(sym => ({name: sym.name, sym}));
+      this.forwardDeclare(decl.moduleSpecifier, namedSyms, false);
```
Comment (evmar) [inline src/tsickle.ts]: Does this mean any module that imports @angular/core ends up with a forwardDeclare line per function in the angular API?

File: test_files/import_prefixed/import_prefixed_mixed.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import * as exporter from './exporter';
+
+let someVar: exporter.TypeExport;
```
Comment (evmar) [inline test_files/import_prefixed/import_prefixed_mixed.ts]: Comment in these files about what property you're looking for.

File: test_files/type_and_value/type_and_value.js [hunk 5]
```diff
@@ -7,13 +7,14 @@ goog.module('test_files.type_and_value.type_and_value');var module = module || {
  */
 
 var conflict = goog.require('test_files.type_and_value.module');
+const tsickle_forward_declare_1 = goog.forwardDeclare("test_files.type_and_value.module");
```
Comment (evmar) [inline test_files/type_and_value/type_and_value.js]: I wonder if the relative order of goog.require / goog.forwardDeclare matters.

File: src/tsickle.ts (hunk 3 above)
Comment (mprobst) [reply src/tsickle.ts]: No. We emit one `forwardDeclare` (as you can see in the test), but then set all symbols exported by that module to be aliased to `tsickle_forward_declare_X.TheSymbolName`.

File: test_files/type_and_value/type_and_value.js (hunk 5 above)
Comment (mprobst) [reply test_files/type_and_value/type_and_value.js]: I don't think so. Fingers crossed.

File: test_files/import_prefixed/import_prefixed_mixed.ts (hunk 4 above)
Comment (mprobst) [reply test_files/import_prefixed/import_prefixed_mixed.ts]: Done.

**Pull request merged**